timemachine/potentials/bonded.py

```diff
@@ -1,18 +1,26 @@
 import jax.numpy as np
 
-def centroid_restraint(conf, params, box, lamb, masses, group_a_idxs, group_b_idxs, kb, b0):
+def centroid_restraint(conf, params, box, lamb, group_a_idxs, group_b_idxs, kb, b0):
 
     xi = conf[group_a_idxs]
     xj = conf[group_b_idxs]
 
-    avg_xi = np.average(xi, axis=0, weights=masses[group_a_idxs])
-    avg_xj = np.average(xj, axis=0, weights=masses[group_b_idxs])
+    avg_xi = np.mean(xi, axis=0)
+    avg_xj = np.mean(xj, axis=0)
 
     dx = avg_xi - avg_xj
-    dij = np.sqrt(np.sum(dx*dx))
+    d2ij = np.sum(dx*dx)
+    d2ij = np.where(d2ij == 0, 0, d2ij) # stabilize derivative
+    dij = np.sqrt(d2ij)
     delta = dij - b0
 
-    return kb*delta*delta
+    # when b0 == 0 and dij == 0
+    return np.where(
+        b0 == 0,
+        kb * d2ij,
```

Comment:
Here `kb = 0.5 * force_constant` as an optimization?

(I note that [in `restraint`](https://github.com/proteneer/timemachine/blob/6e9a150bca9154442d924e4a6486a39ef27bf46a/timemachine/potentials/bonded.py#L74) the same convention is used but [in `harmonic_bond`](https://github.com/proteneer/timemachine/blob/6e9a150bca9154442d924e4a6486a39ef27bf46a/timemachine/potentials/bonded.py#L135) the `kb = force_constant` convention is used. )

Reply:
I don't think there's a consistent form here. OpenMM Harmonic Bond and Angles both use 0.5*k, but periodic torsions use 1*k, and AMBER consistently uses 1*k across the board, so I just chose the simplest convention.